elrond-wasm/src/storage/mappers/linked_list_mapper.rs

```diff
@@ -9,36 +9,58 @@ use crate::{
 use alloc::vec::Vec;
 use core::marker::PhantomData;
 use elrond_codec::{
-    elrond_codec_derive::{TopDecode, TopDecodeOrDefault, TopEncode, TopEncodeOrDefault},
-    DecodeDefault, EncodeDefault, TopDecode, TopEncode,
+    elrond_codec_derive::{
+        NestedDecode, NestedEncode, TopDecode, TopDecodeOrDefault, TopEncode, TopEncodeOrDefault,
+    },
+    DecodeDefault, EncodeDefault, NestedDecode, NestedEncode, TopDecode, TopEncode,
 };
+use storage_get::storage_get_len;
 
 const NULL_ENTRY: u32 = 0;
 const INFO_IDENTIFIER: &[u8] = b".info";
-const NODE_IDENTIFIER: &[u8] = b".node_links";
-const VALUE_IDENTIFIER: &[u8] = b".value";
+const NODE_IDENTIFIER: &[u8] = b".node";
+
+#[derive(NestedEncode, NestedDecode, TopEncode, TopDecode, PartialEq, Clone, Copy)]
+pub struct LinkedListNode<T: NestedEncode + NestedDecode + TopEncode + TopDecode + Clone> {
+    pub(crate) value: T,
+    pub(crate) node_id: u32,
+    pub(crate) next_id: u32,
+    pub(crate) prev_id: u32,
+}
+
+impl<T: NestedEncode + NestedDecode + TopEncode + TopDecode + Clone> LinkedListNode<T> {
+    pub fn get_value(&self) -> T {
+        self.value.clone()
+    }
```

Comment:
Most of the time, developers won't care about any field other than the `value`, so cloning on every `get` is an unneeded performance penalty, especially if we're dealing with managed types.

Keep this method as it is (aside from renaming), but also add another two (note: not 100% sure about the syntax here, please check if it compiles)
```suggestion
    pub fn get_value_cloned(&self) -> T {
        self.value.clone()
    }
    
    pub fn get_value_as_ref<'a>(&'a self) -> &'a T {
        &self.value
    }
    
    pub fn into_value(self) -> T {
        self.value
    }
    
```